sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1454,8 +1456,10 @@ pi_result _pi_queue::executeCommandList(pi_command_list_ptr_t CommandList,
       // command-list itself. This host-visible event will not be
       // waited/released by SYCL RT, so it must be destroyed after all events in
       // the batch are gone.
-      PI_CALL(piEventRelease(HostVisibleEvent));
-      PI_CALL(piEventRelease(HostVisibleEvent));
+      // We know here that refcount is more than 2 so decrement refcount
+      // manually to avoid calling piEventRelease under queue lock.
+      assert(HostVisibleEvent->RefCount > 2);
+      HostVisibleEvent->RefCount -= 2;
```

Comment:
That would make traces much harder to read as the decrement wouldn't be shown there.

Reply:
I agree, fixed.